src/scripts/main.js

```diff
@@ -1,3 +1,248 @@
+
+tHeadCells.forEach((cell, index) => {
+  cell.addEventListener('click', () => {
+    const receivedRows = sortByIdAndType(index);
+    receivedRows.forEach(row => tbody.append(row));
+  })
+})
+
+function sortByIdAndType(index) {
+  const currentRows = [...tbody.querySelectorAll('tr')];
+  const rows = [...currentRows];
+
+  rows.sort((rowA, rowB) => {
+    const cellA = rowA.cells[index].innerText;
+    const cellB = rowB.cells[index].innerText;
+
+    return cellA.localeCompare(cellB, 'en', { numeric: true });
+  });
+
+  if (JSON.stringify(rows.map(row => row.innerText))
+    === JSON.stringify(currentRows.map(row => row.innerText))) {
+    rows.sort((rowA, rowB) => {
+      const cellA = rowA.cells[index].innerText;
+      const cellB = rowB.cells[index].innerText;
+
+      return cellB.localeCompare(cellA, 'en', { numeric: true });
+    });
+  }
+  return rows;
+}
+
+tbody.addEventListener('click', (event) => {
+  if (event.target.parentElement.tagName !== 'TR') {
+    return;
+  }
+
+  const activeClasses = tbody.querySelectorAll('.active');
+  activeClasses.forEach((activeClass) => {
+    activeClass.classList.remove('active');
+  })
+
+  event.target.parentElement.className = 'active';
+})
+
+tbody.addEventListener('dblclick', (event) => {
+  if (event.target.tagName !== 'TD') {
+    return;
+  }
+
+  const defaultValue = event.target.textContent.trim();
+
+  const inputElement = document.createElement('input');
+  inputElement.type = 'text';
+  inputElement.classList.add('cell-input');
+  inputElement.style.width = defaultValue.length * 10 + 'px';
+  inputElement.value = defaultValue;
+
+  event.target.textContent = '';
+  event.target.append(inputElement);
+  inputElement.focus();
+
+  inputElement.addEventListener('blur', () => {
+    if (inputElement.value === '') {
+      inputElement.parentElement.textContent = defaultValue;
+    } else {
+      inputElement.parentElement.textContent = inputElement.value;
+    }
+  })
+
+  inputElement.addEventListener('keypress', (event) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      if (inputElement.value === '') {
+        inputElement.parentElement.textContent = defaultValue;
+      } else {
+        inputElement.parentElement.textContent = inputElement.value;
+      }
+    }
+  })
+})
+
+function employeeForm() {
+  const form = document.createElement('form');
+  form.classList.add('new-employee-form');
+
+  const nameInput = createInput('name');
+  createLabel('Name:', nameInput, form);
+
+  const positionInput = createInput('position');
+  createLabel('Position:', positionInput, form);
+
+  const officeSelect = createSelect('office', officeOptions);
+  createLabel('Office:', officeSelect, form);
+
+  const ageInput = createInput('age', 'number');
+  createLabel('Age:', ageInput, form);
+
+  const salaryInput = createInput('salary', 'number');
+  createLabel('Salary:', salaryInput, form);
+
+  const saveButton = document.createElement('button');
```

Comment:
just as a recommendation. You can try to apply [insertAdjacentHTML](https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentHTML) instead of creating each element by your own